### PR TITLE
refactor: extract TradeConfirmation component from LimitOrdersConfirmModal

### LINKED ISSUE #2429: Create a generic component TradeConfirmationModal (to replace LimitOrdersConfirmModal)
image

For Advanced orders, we will need a component for the confirmation modal.
Need to refactor `LimitOrdersConfirmModal` to create a new universal solution for confirming any trade.

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2429

Refactored `LimitOrdersConfirmModal` in order to create a generic `TradeConfirmation` component that can be used in any trade widget.
Unfortunately, in the changes we can't see that the component is just refactored version of `LimitOrdersConfirmModal`. But, in [the first commit](https://github.com/cowprotocol/cowswap/pull/2564/commits/bc45682d6c0e7d2ce47f711808a4d2ed0a445a92) you can see that it just was moved using `git mv`.

`TradeConfirmation` looks like this:

image

Essentially, it's just a representation of sell + buy amounts with an arbitrary content at the bottom.

  ## Changes
 - [refactor: move LimitOrdersConfirm to modules/trade](https://github.com/cowprotocol/cowswap/pull/2564/commits/bc45682d6c0e7d2ce47f711808a4d2ed0a445a92)
 - [refactor: rename LimitOrdersConfirm to TradeConfirmation](https://github.com/cowprotocol/cowswap/pull/2564/commits/1efab7abd337e8fd18bb483bbdb01459b5758db6)
 - [refactor: extract limit-orders specific code from TradeConfirmation](https://github.com/cowprotocol/cowswap/pull/2564/commits/63ea051d328e71b699e891310b7c90c30a401f5b)
 - [refactor: narrow down CurrencyAmountPreview props](https://github.com/cowprotocol/cowswap/pull/2564/commits/9b3ec1c69ccbd393f780f4f498eb43d4a3e19811)

### CODE REVIEW

File: src/common/pure/CurrencySelectButton/index.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+import { Nullish } from 'types'
 
 import { CurrencyLogo } from 'common/pure/CurrencyLogo'
 import { TokenSymbol } from 'common/pure/TokenSymbol'
 
 import * as styledEl from './styled'
 
 export interface CurrencySelectButtonProps {
-  currency?: Currency
+  currency?: Nullish<Currency>
```
Comment (anxolin) [inline src/common/pure/CurrencySelectButton/index.tsx:12]: do you still need the question-mark in the prop?

Comment (shoom3301) [reply src/common/pure/CurrencySelectButton/index.tsx:12]: Yes. `Optional !== undefined`

Comment (anxolin) [reply src/common/pure/CurrencySelectButton/index.tsx:12]: fine with me, different is subtle, the point is if we want to make it optional.

Currency is even in the name of the component, and the 2 uses provide the prop

https://github.com/cowprotocol/cowswap/blob/37d4879d597d7e59df1511eb867a626ef7419d0b/src/common/pure/CurrencyInputPanel/CurrencyInputPanel.tsx#L137

https://github.com/cowprotocol/cowswap/blob/37d4879d597d7e59df1511eb867a626ef7419d0b/src/modules/limitOrders/pure/ReceiptModal/CurrencyField.tsx#L20

**Review by anxolin — APPROVED** [review]
Great changes! just reviewed the code, i didn't test the app itself

**Review by nenadV91 — APPROVED** [review]
LGTM

File: src/common/pure/CurrencySelectButton/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/common/pure/CurrencySelectButton/index.tsx:12]: Aaaah, got your point. Will refactor it in the next PR.
Thank you!

**Comment by elena-zh** [comment]
In this PR (or maybe some earlier) it is impossible to create a limit order with a native token in the To field
image

**Pull request merged**